src/MOI_wrapper.jl

```diff
@@ -73,6 +73,7 @@ mutable struct Optimizer{OT <: MOI.ModelLike} <: MOI.AbstractOptimizer
     dual_optimal::Vector{Union{Vector{Float64}, Float64}}  # refer - https://github.com/jump-dev/DiffOpt.jl/issues/21#issuecomment-651130485
     var_idx::Vector{VI}
     con_idx::Vector{CI}
+    gradient_cache::NamedTuple
```

Comment:
Why not using `Union{Nothing, NamedTuple{...}}` or even `Union{Nothing, Cache}` with an actual struct ? Currently, it's type unstable

Reply:
This will need at least `Union{Nothing, NamedTuple{T1}, NamedTuple{T2}}` for the cache, since we have possible cache types depending on conic or not

Reply:
I'd still prefer `struct Cache1 end; struct Cache2 end; Union{Nothing, Cache1, Cache2}`, otherwise it's not so easy to follow what's going on.
Moreover, it seems to me that `NamedTuple` are used when you don't know a priori what is the struct so it's like creating new structs on-the-fly why here we know very well the struct at coding time so we should use structs as it's more readable.